app/views/RoomMembersView/helpers.ts

```diff
@@ -89,8 +89,13 @@ export const navToDirectMessage = async (item: IUser, isMasterDetail: boolean):
 				handleGoRoom({ rid: result.room?._id as string, name: item.username, t: SubscriptionType.DIRECT }, isMasterDetail);
 			}
 		}
-	} catch (e) {
+	} catch (e: any) {
 		log(e);
+		if (e?.data?.details?.method === 'createDirectMessage' && e?.data?.errorType === 'error-not-allowed') {
+			EventEmitter.emit(LISTENER, {
+				message: I18n.t('error-action-not-allowed', { action: I18n.t('Create_Direct_Messages') })
```

Comment:
`'error-action-not-allowed'` isn't being used anywhere else.
Maybe we should remove it?

Recently on [this PR](https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/5356), we implemented a similar feature as the one you're proposing.
Can you make yours similar to this? https://github.com/RocketChat/Rocket.Chat.ReactNative/blob/3e9ab8877c03b263b92bc6ade985ac2008fdbff6/app/containers/ActionSheet/Item.tsx#L26

Reply:
Sure, sir